app/javascript/packages/document-capture/components/address-search.tsx

```diff
@@ -183,13 +185,15 @@ function AddressSearch({
     spinnerButtonRef.current?.toggleSpinner(isLoading);
   }, [isLoading]);
 
-  useEffect(() => {
-    locationResults && onFoundLocations(locationResults);
+  useDidUpdateEffect(() => {
+    onFoundLocations(locationResults);
+
     foundAddress && onFoundAddress(foundAddress);
-  }, [locationResults, foundAddress]);
+  }, [locationResults]);
```

Comment:
Should `foundAddress` stay here? Do we want `onFoundAddress` to be called when `foundAddress` changes?

Reply:
@eileen-nava There's an [exhaustive-deps rule](https://github.com/facebook/react/blob/main/packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js) published by Facebook that if enabled would flag this as a violation b/c `foundAddress` is missing from the hook dependencies.

Generally when the dependencies don't match the values used in a hook like this, it's considered a code smell in React. This is a very likely location for a bug to occur because the value is missing.

Reply:
How about we split this into two effects, one for `locationResults`/`onFoundLocations` and a separate one for `foundAddress`/`onFoundAddress`?

Reply:
That way we can both follow the rule and avoid the updates.

Reply:
`foundAddress` is the first entry from `locationResults`, so it shouldn't fire an update before the results are available if we keep the same conditional logic.

Reply:
Oh, look like I misread the code. In that case you'd need to make the conditional check both `foundAddress` and `locationResults` before  triggering the `onFoundAddress` callback.

Reply:
I think I see what you're trying to do, though. You'd need a ref or similar to track if the `foundAddress` was associated with the new or previous `locationResults`.

Reply:
Reading through the results, I think you may be able to get similar behavior to what you're looking for by checking the `isLoading` flag before calling `onFoundAddress`. E.g.:

```typescript
  useDidUpdateEffect(() => {
    if (isLoading) {
      return;
    }
    
    onFoundLocations(locationResults);

    foundAddress && onFoundAddress(foundAddress);
  }, [locationResults, foundAddress, isLoading]);
```

Reply:
To second what Matt is saying, if I add `foundAddress` to the dependency array without adding back in some sort of conditional logic, it causes the address to update before the locations update.  You can see this bug in the “before video” for scenario 2.  

I'm not against adding the conditional logic back in and re-adding the `foundAddress` dependency to the dependency array.  I’m mostly concerned about getting this PR approved and merged by sprint’s end.

Reply:
I discussed this with Matt in Slack and I am also going to share here for posterity.  

When I set the dependencies array to `[locationResults, foundAddress, isLoading]`, I found a bug where the first search’s results would display before the second search’s results.  

I poked through debugger and found that `isLoading` incorrectly equaled `true` before both the second search’s address candidates and usps locations had loaded.  (I think this has to do with swr caching?  More investigation is needed.)  This caused the `useDidUpdateEffect` hook to incorrectly set the `foundAddress` and `locationResults`, even when there was a conditional like

```
    if (isLoading) {
      return;
    }
```

The plan is to 
A) merge working code that doesn’t have exhaustive dependencies
B) make a ticket to satisfy the exhaustive dependencies requirement and investigate what’s going on with `isLoading`

Reply:
> How about we split this into two effects, one for locationResults/onFoundLocations and a separate one for foundAddress/onFoundAddress?

When I tried that, it made the updated address display before the updated locations results.  That contradicts the ticket's AC.